File: src/content/design-patterns/factory/js/code.ts
const enum PaymentSystemType {
  PayPal = "paypal",
  WebMoney = "webmoney",
  CatBank = "cat-bank",
}







export function case1(
  type: PaymentSystemType,
  base: number,
  amount: number
): number {
  throw new Error("Not implemented");
}

const enum PaymentSystemActionType {
  Payment = "payment",
  Payout = "payout",
}







export function case2(
  type: PaymentSystemType,
  base: number,
  action: PaymentSystemActionType,
  amount: number
): number {
  throw new Error("Not implemented");
}

File: src/content/design-patterns/factory/python/code.py
from enum import Enum

class PaymentSystemType(Enum):
  PAYPAL = "paypal"
  WEBMONEY = "webmoney"
  CAT_BANK = "cat-bank"







def case1(
  type: PaymentSystemType,
  base: int,
  amount: int
) -> int:
  raise NotImplementedError

class PaymentSystemActionType(Enum):
  PAYMENT = "payment"
  PAYOUT = "payout"








def case2(
  type: PaymentSystemType,
  base: int,
  action: PaymentSystemActionType,
  amount: int
) -> int:
  raise NotImplementedError